apps/explorer/src/components/Object/ObjectFieldsCard.tsx

```diff
@@ -5,31 +5,36 @@ import { useGetObject } from '@mysten/core';
 import { useNormalizedMoveStruct } from '@mysten/dapp-kit';
 import { Search24 } from '@mysten/icons';
 import { Text, LoadingIndicator, Combobox, ComboboxInput, ComboboxList } from '@mysten/ui';
-import { useState } from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
 
 import { FieldItem } from './FieldItem';
 import { ScrollToViewCard } from './ScrollToViewCard';
 import { getFieldTypeValue } from './utils';
+import { FieldsCard, FieldCollapsible, FieldsContainer } from '~/components/Object/FieldsUtils';
 import { Banner } from '~/ui/Banner';
-import { DisclosureBox } from '~/ui/DisclosureBox';
-import { TabHeader } from '~/ui/Tabs';
-import { ListItem, VerticalList } from '~/ui/VerticalList';
+import { DescriptionItem } from '~/ui/DescriptionList';
 
 interface ObjectFieldsProps {
 	id: string;
+	setCount: (count: number) => void;
```

Comment:
This seems like a very bad way to do this. why can't the parent just fetch this data?

Reply:
removed this to call the data from parent